presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -772,7 +772,7 @@ public void testDistinctMultipleFields()
     public void testDistinctJoin()
             throws Exception
     {
-        assertQuery("SELECT COUNT(DISTINCT b.quantity), a.orderstatus " +
+        assertQuery("SELECT COUNT(DISTINCT CAST(b.quantity AS BIGINT)), a.orderstatus " +
```

Comment:
This commit is just a partial cherry pick of https://github.com/prestodb/presto/commit/dbbe14e7f974b80aba11b4a45695ec81e84545e2 from master. I'm not quite sure what exactly changes were introduced with TPCH 0.6.